app/src/features/surveys/view/SurveyAnimals.tsx

```diff
@@ -101,24 +150,43 @@ const SurveyAnimals: React.FC = () => {
     }
   };
 
-  const handleTelemetrySave = async (survey_critter_id: number, data: IAnimalTelemetryDevice) => {
+  const handleTelemetrySave = async (survey_critter_id: number, data: IAnimalTelemetryDevice[]) => {
     const critter = critterData?.find((a) => a.survey_critter_id === survey_critter_id);
-    const critterTelemetryDevice = { ...data, critter_id: critter?.critter_id ?? '' };
-    try {
-      await bhApi.survey.addDeployment(projectId, surveyId, survey_critter_id, critterTelemetryDevice);
-    } catch (e) {
-      dialogContext.setSnackbar({
-        open: true,
-        snackbarMessage: (
-          <Typography variant="body2" component="div">
-            {`Could not add deployment.`}
-          </Typography>
-        )
-      });
-    } finally {
-      setOpenAddDeviceDialog(false);
-      refreshDeployments();
+    const critterTelemetryDevice = { ...data[0], critter_id: critter?.critter_id ?? '' };
+    if (telemetryFormMode === 'add') {
+      try {
+        await bhApi.survey.addDeployment(projectId, surveyId, survey_critter_id, critterTelemetryDevice);
+        setPopup(`Successfully added deployment.`);
+      } catch (e) {
+        setPopup(`Failed to add deployment.`);
+      }
+    } else if (telemetryFormMode === 'edit') {
+      try {
+        //Better to do a high level try catch or try catch calls individually? Not sure.
```

Comment:
Separate try-catch is fine, especially if they are catching and rethrowing distinct errors. However, the code at the bottom (`setOpenDeviceDialog(false); refreshDeployments();`) will not run if the Add case rethrows an error